core/src/test/java/com/linecorp/armeria/client/HttpClientMaxConcurrentStreamTest.java

```diff
@@ -168,4 +170,143 @@ public void connectionClosed(SessionProtocol protocol, InetSocketAddress remoteA
             assertThat(closes).hasValue(0);
         }
     }
+
+    @Test
+    public void succeedWhenExceedMaxStreams() throws Exception {
+        final WebClient client = WebClient.builder(server.uri(SessionProtocol.H2C, "/"))
+                                          .factory(clientFactory)
+                                          .build();
+        final List<CompletableFuture<AggregatedHttpResponse>> receivedResponses = new ArrayList<>();
+
+        // running inside event loop ensures requests are queued before initial connect completes.
+        for (int j = 0; j < MAX_CONCURRENT_STREAMS + 1; j++) {
+            receivedResponses.add(client.get(PATH).aggregate());
+        }
+
+        await().untilAsserted(() -> assertThat(responses).hasSize(MAX_CONCURRENT_STREAMS + 1));
```

Comment:
also add checks for `pendingAcquisitionMillis` to ensure pendingAcquisition was really used

Reply:
done -- not applying to test for multipleEventLoops since this check depends on how event loops are acquired. (this would actually be do-able, but not sure if it's worth it)